src/librustc/session/mod.rs

```diff
@@ -176,14 +176,15 @@ impl Session {
     pub fn abort_if_errors(&self) {
         self.diagnostic().abort_if_errors();
     }
-    pub fn abort_if_new_errors<F>(&self, mut f: F)
-        where F: FnMut()
+    pub fn abort_if_new_errors<F, T>(&self, f: F) -> T
```

Comment:
(and the other problem with `fn try_anything` is that it may imply to some that `try!` or `Result` is somehow involved, which really isn't case here. I'd be okay with `fn call_and_abort_on_new_errors` or something along those lines.)